backend/tests/prisonerProfileService.test.ts

```diff
@@ -868,4 +884,260 @@ describe('prisoner profile service', () => {
+      dataComplianceApi.getOffenderRetentionRecord.mockReturnValue({})
+      // @ts-expect-error ts-migrate(2339) FIXME: Property 'getPomByOffenderNo' does not exist on ty... Remove this comment to see the full error message
+      allocationManagerApi.getPomByOffenderNo.mockResolvedValue({ primary_pom: { name: 'SMITH, JANE' } })
+      // @ts-expect-error ts-migrate(2322) FIXME: Type 'undefined[]' is not assignable to type 'stri... Remove this comment to see the full error message
+      config.apis.complexity.enabled_prisons = []
+      // @ts-expect-error ts-migrate(2339) FIXME: Property 'getLearnerNeurodivergence' does not exist on ty... Remove this comment to see the full error message
+      curiousApi.getLearnerNeurodivergence.mockResolvedValue([neurodivergenceDataNoSupport])
+    })
+
+    it('should return false when user has no neurodiversity data', async () => {
+      await service.getPrisonerProfileData(context, offenderNo)
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const neurodivergence = await curiousApi.getLearnerNeurodivergence(context, offenderNo)
+      const hasDivergenceSupport = Boolean(neurodivergence[0]?.neurodivergenceSupport?.length > 0)
+
+      // @ts-expect-error ts-migrate(2339) FIXME: Property 'currentUser' does not exist on type '{}'... Remove this comment to see the full error message
+      expect(oauthApi.currentUser).toHaveBeenCalledWith(context)
+      expect(hasDivergenceSupport).toEqual(false)
+    })
+    it('should return false when no neurodiversity support needs are identified', async () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      curiousApi.getLearnerNeurodivergence.mockResolvedValue([neurodivergenceDataNoSupportIdentified])
+      await service.getPrisonerProfileData(context, offenderNo)
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const neurodivergence = await curiousApi.getLearnerNeurodivergence(context, offenderNo)
+      const hasDivergenceSupport = neurodivergence.some((element) => {
+        // eslint-disable-next-line eqeqeq
+        return !(element.neurodivergenceSupport == NeurodivergenceType.NoidentifiedNeurodiversityNeed)
+      })
+      // @ts-expect-error ts-migrate(2339) FIXME: Property 'currentUser' does not exist on type '{}'... Remove this comment to see the full error message
+      expect(oauthApi.currentUser).toHaveBeenCalledWith(context)
+      expect(hasDivergenceSupport).toEqual(false)
+    })
+    it('should handle the not found error', async () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      curiousApi.getLearnerNeurodivergence.mockResolvedValue([neurodivergenceNotFound])
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const errorResponse = await curiousApi.getLearnerNeurodivergence(context, offenderNo)
```

Comment:
This test is testing prisonerProfileService - so it needs to call prisonerProfileService methods.

This applies to all the tests you added.